common/persistence/persistenceInterface.go

```diff
@@ -60,7 +60,20 @@ type (
 	}
 
 	// TaskStore is a lower level of TaskManager
-	TaskStore = TaskManager
+	TaskStore interface {
+		Closeable
+		GetName() string
+		CreateTaskQueue(request *InternalCreateTaskQueueRequest) error
+		GetTaskQueue(request *InternalGetTaskQueueRequest) (*InternalGetTaskQueueResponse, error)
+		ExtendLease(request *InternalExtendLeaseRequest) error
+		UpdateTaskQueue(request *UpdateTaskQueueRequest) (*UpdateTaskQueueResponse, error)
+		ListTaskQueue(request *ListTaskQueueRequest) (*ListTaskQueueResponse, error)
+		DeleteTaskQueue(request *DeleteTaskQueueRequest) error
+		CreateTasks(request *CreateTasksRequest) (*CreateTasksResponse, error)
+		GetTasks(request *GetTasksRequest) (*GetTasksResponse, error)
+		CompleteTask(request *CompleteTaskRequest) error
+		CompleteTasksLessThan(request *CompleteTasksLessThanRequest) (int, error)
```

Comment:
maybe worth considering making this `RangeCompleteTasks` allowing specifying [inclusive, exclusive) task IDs